cas-server-support-osf/src/main/java/io/cos/cas/adaptors/postgres/handlers/OpenScienceFrameworkInstitutionHandler.java

```diff
@@ -60,16 +60,45 @@ public String findInstitutionLogoutUrlById(final String id) {
     }
 
     /**
-     * Return a map of institution name and login url.
+     * Check if a delegation client is indeed an institution one with a matching protocol.
      *
-     * @param target The osf service target after successful institution login
-     * @return Map&lt;String, String&gt;
+     * @param clientName the name of the client
+     * @return true if the client is institution, false otherwise
+     */
+    public boolean isDelegatedInstitutionLogin(final String clientName) {
+        final OpenScienceFrameworkInstitution institution
+                = openScienceFrameworkDao.findOneInstitutionById(clientName.toLowerCase());
```

Comment:
I think I misunderstood this before. Here is the `toLowerCase()` comparison for `clientName` as `institutionId`.